cmd/frontend/webhooks/webhooks_test.go

```diff
@@ -0,0 +1,138 @@
+package webhooks
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/sourcegraph/log/logtest"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbtest"
+	"github.com/sourcegraph/sourcegraph/internal/encryption/keyring"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWebhooksHandler(t *testing.T) {
+	logger := logtest.Scoped(t)
+	db := database.NewDB(logger, dbtest.NewDB(logger, t))
+	u, err := db.Users().Create(context.Background(), database.NewUser{
+		Email:           "test@user.com",
+		Username:        "testuser",
+		EmailIsVerified: true,
+	})
+	require.NoError(t, err)
+	dbWebhooks := db.Webhooks(keyring.Default().WebhookKey)
+	gitLabWH, err := dbWebhooks.Create(
+		context.Background(),
+		extsvc.KindGitLab,
+		"http://gitlab.com",
+		u.ID,
+		types.NewUnencryptedSecret("somesecret"))
+	require.NoError(t, err)
+
+	gitHubWH, err := dbWebhooks.Create(
+		context.Background(),
+		extsvc.KindGitHub,
+		"http://github.com",
+		u.ID,
+		types.NewUnencryptedSecret("githubsecret"),
+	)
+
+	require.NoError(t, err)
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		handler := NewHandler(db)
+
+		handler.ServeHTTP(w, r)
+	}))
```

Comment:
Haven't checked, but doesn't something like this work?

```suggestion
	srv := httptest.NewServer(NewHandler(db))
```

Reply:
It does indeed :) 🙌